src/main/java/com/ita/if103java/ims/service/impl/WarehouseItemAdviceServiceImpl.java

```diff
@@ -60,24 +61,27 @@ public WarehouseItemAdviceDto getAdvice(Long itemId, UserDetailsImpl userDetails
         final User user = userDetails.getUser();
         final List<WeightAssociateDto> suppliers = bestTradeService.findBestSuppliersByItemId(itemId);
         final List<WeightAssociateDto> clients = bestTradeService.findBestClientsByItemId(itemId);
-        final List<WarehouseToAssociateDistanceDto> warehouseAssociateDistances = keepIfAvailableRoute(
+        final List<Long> supplierIds = getAssociateIds(suppliers);
+        final List<Long> clientIds = getAssociateIds(clients);
+        final List<Long> topWarehouseIds = topWarehouseDao.findAllIds(user.getAccountId());
+
+        if (supplierIds.isEmpty() || clientIds.isEmpty() || topWarehouseIds.isEmpty()) {
+            throw new ImpossibleWarehouseAdviceException("Your account doesn't have enough valuable info to provide an advice");
```

Comment:
Which `HTTP status code` should be returned in that case?
What do you think, @ikrasWork ?

Reply:
Hard to answer
Let it be 200